warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeDetectionVisitor.java

```diff
@@ -28,29 +38,45 @@ public static boolean mustExpandBoundedRange(ShardQueryConfiguration config, Met
     }
     
     @Override
-    public Object visit(ASTERNode node, Object data) {
-        if (null != data) {
-            AtomicBoolean hasBounded = (AtomicBoolean) data;
-            hasBounded.set(true);
+    public Object visit(ASTReference node, Object data) {
+        if (BoundedRange.instanceOf(node)) {
+            LiteralRange range = JexlASTHelper.findRange().getRange(node);
+            try {
+                if (helper.getNonEventFields(config.getDatatypeFilter()).contains(range.getFieldName())) {
+                    if (null != data) {
+                        AtomicBoolean hasBounded = (AtomicBoolean) data;
+                        hasBounded.set(true);
+                    }
+                }
+            } catch (TableNotFoundException e) {
+                throw new DatawaveFatalQueryException("Cannot access metadata", e);
+            }
+            
+            return false;
+        } else {
+            return super.visit(node, data);
         }
-        
-        return false;
-        
     }
     
     @Override
-    public Object visit(ASTNRNode node, Object data) {
-        if (null != data) {
-            AtomicBoolean hasBounded = (AtomicBoolean) data;
-            hasBounded.set(true);
+    public Object visit(ASTERNode node, Object data) {
```

Comment:
shouldn't these be LT/LTE and GT/GTE in order to detect a bounded range?

Reply:
All I did here was replace the visit(ASTLTNode) and visit(ASTGTNode) with the detection of the BoundedRange marker in visit(ASTReference).  If you look where this thing is used, it is also looking for ER and NR nodes.  I do concur that the initial implementation was missing LE and GE node checks.  By changing this to look for BoundedRange markers, I essentially have made that change.